src/api/pipelineRuns.test.js

```diff
@@ -326,3 +328,226 @@ it('startPipelineRun', () => {
     }
   );
 });
+
+it('generate new pipeline run minimum for rerun', () => {
```

Comment:
I think it would be better to replace `minimum` and `maximum` in these tests with descriptions of what they're actually doing, e.g. 'with labels, annotations, and params' or whatever

This gives more context in test reports etc. without having to revert to reading the code to understand what the test is doing.

Reply:
- `rerun. generate new pipeline run with spec, metadata - name and namespace`
- `rerun. generate new pipeline run with spec, status, metadata - annotations, labels, uid and resourceVersion`

vs

- `rerun. generate new pipeline run. minimum`
- `rerun. generate new pipeline run. maximum`

Here is the rational behind `maximum`
- nobody would bother to read such long description (with listing words)
- nobody would bother to change such description on changes, which makes description incorrect and more harmful
- `maximum` assumed to be used by anybody making changes there and naming helps to convince others about that (sure, not 100% convincible)

Here is the rational behind `Minimum`
- making sure that this is a minimum input and in the absence of other fields function doesn't fail
- by word `minimum` it prevents others from making unrelated modifications in this test (sure, not 100% convincible)

Reply:
> only few would change such description on changes, which makes description incorrect and more harmful

I agree this could be a problem, I shouldn't have listed the affected fields in my example, I was just trying to give an example that explains what it's doing, e.g. "… with all processed fields", perhaps that is a better description. Also, it is up to the maintainers to ensure that if the test changes the test and its description still match / make sense.

>  prevents others from making unrelated modifications in this test

Not sure about this, also as above it would be the responsibility of the maintainers to identify this and provide feedback if it happens.

I still think we can be clearer than 'minimum' and 'maximum' here but I won't block the PR on this.

Reply:
Changed to
- `edit. generate new pipeline run with all processed fields`
- `edit. generate new pipeline run with minimum possible fields`

Another option would be to switch to more business terms, e.g.
- generate new pipeline run from completed pipeline run
at the same time I don't provide 100% of fields (however I can copy-paste from any completed pipeline run and that would make it very close to that)
- generate new pipeline run from non-created pipeline run with minimum fields